state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1216,35 +1228,47 @@ pub struct PoolLiquidity {
 	pub range_orders: Vec<RangeOrderLiquidity>,
 }
 
-#[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Deserialize, Serialize)]
+#[derive(
+	Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Deserialize, Serialize, MaxEncodedLen,
+)]
 pub struct UnidirectionalSubPoolDepth {
 	/// The current price in this sub pool, in the given direction of swaps.
 	pub price: Option<Price>,
 	/// The approximate amount of assets available to be sold in the specified price range.
 	pub depth: Amount,
 }
 
-#[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Deserialize, Serialize)]
+#[derive(
+	Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Deserialize, Serialize, MaxEncodedLen,
+)]
 pub struct UnidirectionalPoolDepth {
 	/// The depth of the limit order pool.
 	pub limit_orders: UnidirectionalSubPoolDepth,
 	/// The depth of the range order pool.
 	pub range_orders: UnidirectionalSubPoolDepth,
 }
 
-#[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Serialize, Deserialize)]
+pub const MAXIMUM_ORDERS_IN_POOL_ORDERBOOK: u32 = 16384;
+
+#[derive(
+	Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Serialize, Deserialize, MaxEncodedLen,
+)]
 pub struct PoolOrder {
 	pub amount: Amount,
 	pub sqrt_price: SqrtPriceQ64F96,
 }
 
-#[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(
+	Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq, Serialize, Deserialize, MaxEncodedLen,
+)]
 pub struct PoolOrderbook {
-	pub bids: Vec<PoolOrder>,
-	pub asks: Vec<PoolOrder>,
+	pub bids: BoundedVec<PoolOrder, ConstU32<MAXIMUM_ORDERS_IN_POOL_ORDERBOOK>>,
```

Comment:
Ok, curious to know why. My reasoning was that it makes more sense to me to return an error instead of returning the maximum amount of orders because the caller may not be aware of that, which can cause bugs on its end.